agent/src/main/java/com/teamscale/jacoco/agent/store/upload/UploadStoreBase.java

```diff
@@ -94,38 +92,38 @@ protected boolean tryUpload(String xml) {
 		} catch (IOException e) {
 			logger.error("Failed to upload coverage to {}. Probably a network problem", uploadUrl, e);
 			return false;
-		} catch (UploadStoreException e) {
+		} catch (UploaderException e) {
 			logger.error("Failed to upload coverage to {}. The configuration is probably incorrect", uploadUrl, e);
 			return false;
 		}
 	}
 
 	/**
-	 * Creates the zip file to upload which includes the given coverage XML and all
-	 * {@link #additionalMetaDataFiles}.
+	 * Creates the zip file to upload which includes the given coverage XML and all {@link #additionalMetaDataFiles}.
 	 */
-	private byte[] createZipFile(String xml) throws IOException {
-		try (ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream()) {
-			try (ZipOutputStream zipOutputStream = new ZipOutputStream(byteArrayOutputStream);
-				 OutputStreamWriter writer = new OutputStreamWriter(zipOutputStream, "UTF-8")) {
-				fillZipFile(zipOutputStream, writer, xml);
+	private File createZipFile(File coverageFile) throws IOException {
+		File zipFile = createZipFileObject(coverageFile);
+		try (FileOutputStream fileOutputStream = new FileOutputStream(zipFile)) {
+			try (ZipOutputStream zipOutputStream = new ZipOutputStream(fileOutputStream)) {
```

Comment:
since you're not catching exceptions here, you can join the two try-with-resources: `try(A a = ..., B b = ...) { }`